Fixes gravity failure event so it doesn't give gravity to shuttles

anconfuzedrock opened the pull request:
:cl:
bugfix: Gravity failure hazards will not give gravity to shuttles anymore.
/:cl:

fixes #29914 by ensuring the event only screws with gravity in areas that already had it enabled. The former check failed to account for shuttles, which start the game with gravity but very often lose it in transit. Thanks to MistakeNot for figuring out how to fix this.

MuckerMayhem reviewed (changes requested):
Appears you used spaces instead of tabs in the indentation, fix that and you're good.

anconfuzedrock commented:
think I got it.

MuckerMayhem commented:
Squash your changes then you're good.

anconfuzedrock commented:
more stuff is broken now since squashing always goes wrong no matter how long I spent trying to figure it out, hang on

anconfuzedrock committed:
stops gravity being enabled in gravityless shuttles

changes [a] to +=, adds newline

adds a newline for real this time

changes true to a

anconfuzedrock commented:
I _think_ I got it.

Spookerton commented on code/modules/events/gravity.dm:
```diff
@@ -10,15 +11,15 @@
 	for(var/area/A in world)
-		if(A.z in affecting_z)
+		if((A.z in affecting_z) && A.has_gravity())
+			gravity_status[A] = TRUE
```
```suggestion
			gravity_status += A
```
I can't see a reason to use areas as keys with static values here.

Spookerton commented on code/modules/events/gravity.dm:
```diff
@@ -10,15 +11,15 @@
 				A.gravitychange(gravity_is_on)
-
 		command_announcement.Announce("Gravity generators are again functioning within normal parameters. Sorry for any inconvenience.", "[location_name()] Gravity Subsystem", zlevels = affecting_z)
+	gravity_status.Cut()
```
```suggestion
	gravity_status.Cut()

```
since there's the other change request - newline at end of file

Spookerton reviewed (changes requested):
and squanch

anconfuzedrock commented:
Donezo to both those, unless I've gone completely insane and I'm hallucinating that I successfully squanched.

MistakeNot4892 commented on code/modules/events/gravity.dm:
```diff
@@ -10,15 +11,16 @@
 	for(var/area/A in world)
-		if(A.z in affecting_z)
+		if((A.z in affecting_z) && A.has_gravity())
+			gravity_status += TRUE
```
This should be `gravity_status[A] = TRUE`.

Spookerton replied:
Woop, I assumed my request had been done verbatim. `+= A`, not `+= TRUE`.
Not getting why using A as a key would be preferable?

MistakeNot4892 replied:
As discussed on Discord it's slightly faster than `|=` and eliminates duplicates, which I assumed `in world` would be producing if it iterates per turf rather than over some area list.

anconfuzedrock commented:
I hope I got it.

SierraKomodo commented:
Kicked not-travis.

Silly bearcat map.
